pkg/types/validators.go

```diff
@@ -1,28 +1,133 @@
 package types
 
 import (
+	"context"
+	"time"
+
 	"github.com/mt-sre/addon-metadata-operator/api/v1alpha1"
 	"github.com/operator-framework/operator-registry/pkg/registry"
 )
 
-type ValidateFunc func(mb MetaBundle) ValidatorResult
+// NewValidator returns a configured validator.
+// code is a required parameter identifying the "AMXXXX" identifier for a validator
+// runner is the validation process the validator will run.
+// opts is a slice of additional options which can be configured.
+func NewValidator(code string, runner ValidatorRunner, opts ...ValidatorOption) Validator {
+	val := Validator{
+		Runner: runner,
+		ValidatorConfig: ValidatorConfig{
+			Code: code,
+		},
+	}
+
+	val.Options(opts...)
+
+	return val
+}
 
+// Validator encapsulates display fields and implementation for
+// tasks performed against MetaBundles to validate their data.
 type Validator struct {
-	Name        string
+	Runner ValidatorRunner
+	ValidatorConfig
+}
+
+// Validate runs a validator's Runner against a MetaBundle.
+func (v *Validator) Validate(mb MetaBundle) ValidatorResult {
+	return v.Runner.Run(v.ValidatorConfig, mb)
+}
+
+// Option applies an option to an instance of Validator.
+func (v *Validator) Options(opts ...ValidatorOption) {
+	for _, opt := range opts {
+		opt(&v.ValidatorConfig)
+	}
+}
```

Comment:
Just out of curiosity, any reason to make it have `Validator` as a receiver and not `ValidatorConfig` as receiver instead.

This is what I was thinking

```suggestion
func (vc *ValidatorConfig) Options(opts ...ValidatorOption) {
	for _, opt := range opts {
		opt(vc)
	}
}
```

Reply:
Nope, lol. Probably just refactored that without thinking. Your change makes perfect sense.